js/ai/src/generate/action.ts

```diff
@@ -240,7 +254,12 @@ async function generate(
     ] as MessageData[],
     tools: newTools,
   };
-  return await generateHelper(registry, nextRequest, middleware);
+  return await generateHelper(
+    registry,
+    nextRequest,
+    middleware,
+    toolCallIteration + 1
+  );
```

Comment:
Can we make sure the arguments are in the same order for generate and generateHeper? It's crossing my eyes that in one it's middleware, toolIteration and in the other it's toolIteration, middleware

Reply:
done.